### DISCUSSION TITLE
Reload hub triggers entity events

### DISCUSSION THREAD

**Discussion by juerg-altwegg** [SukramJ/aiohomematic#1629] [Q&A] [answered]
After reloading a Hub within the Homematic(IP) Local Integration, **all** button entities are triggered (perhaps other entities as well, but I use only buttons to trigger automations). Unfortunately, the event type keeps the former value. Thus, for example, an event "short_press" is fired for a light automation and a light turns on. My challange: after a hub reload several lights turns on or off and blinds are closing or opening randomly like the whole house is going crazy :-)
This behavior is reproducible either on a physical CCU3 or on the Home Assistant add-on "RaspberryMatic" (virtual CCU). 
Workaround: Never touch the "reload" function for hubs and restart the complete Home Assistant if something has changed on the CCU configuration.
Any ideas, how to prevent the integration from triggering all those events in case of a hub reload?

**Comment by SukramJ** [comment]
It might be helpful to add relevant code snippets.

**Reply by juerg-altwegg** [reply]
It's all done via the Home Assistant GUI. I'm just an user of this integration...

**Reply by SukramJ** [reply]
Add Screenshots or show the YAML code.

**Reply by juerg-altwegg** [reply]
How to reproduce:
1. Connect the integration Homematic(IP) Local to CCU instance (in this example a virtual RaspberryMatic)
2.  Make sure that at least one HomeMatic button entity is present (here a HM-PB-2-WM55-2)
3. Create a new automation with an entity from this button (here Channel 1); example automation with event "press_short" (screenshots in german lanuage, sorry):
![example_automation](https://github.com/user-attachments/assets/e0bfcb8e-3f5a-4a22-9c57-6eb2976f519c)

4. Reload the integration
![reload](https://github.com/user-attachments/assets/d1570390-2f93-48ce-893d-990201c13dba)

6. Result: the entity will be triggered with event "press_short" despite there was no button pressed and the light bulb is turned on.

**Reply by SukramJ** [reply]
Looks similar to [this](https://github.com/home-assistant/core/issues/115986#issuecomment-2078284174).

Your automation is fired due the fact, that all entites switch to state `unknown` while reloading and back to the `last timestamp fired` afterwards. Additionally the condition of the event_type is met, and your light toggles.
The procedure of restarting HA works different, and automations are activated later.

I have no solution with event entities for you because this stuff is HA only (maybe with a bug), and the integration is not firing anything (otherwise the timestamp would change)

If you just want to do something based on a pushed button i would use [device triggers](https://www.home-assistant.io/docs/automation/trigger/#device-triggers), that can be configured by UI.

**Reply by juerg-altwegg** [reply]
Thank you very much for digging into the deep of the code and the hint with the similar problem of MQTT-Integration. It seems indeed a similar effect caused by HA. I'm afraid this problem will not be solved in the foreseeable future. However, perhaps other users are running into the same obstacles, therefore I deliver some background information:
- Trigger of "ghost" events only occurs by reloading the integration (fortunately, reloading of YAML configs has no effect, )
- In my case, I use the reload function often, because I'm in the process to replace my physical CCU3 complete with HA and virtual RaspberryMatic (HA add-on). Therefore I disconnect sensor by sensor and actor by actor from CCU3 and connect it to the RaspberryMatic. The whole logic for lights, blinds, heatings and smoke detectors will be rebuilt in HA (which is by the way much more reliable and thanks to blueprints much clearer than the CCU "programs"). Thus, after each sensor or actor I did a reload cycle to see it in the GUI list of HA. 
- Device triggers are not the recommended way (even if they work in a simple automation like the one above), especially if you are using blueprints (i.e. state machine for the blinds). 
- As a small consolation: After moving all HomeMatic components to the "new home", the need for reloading the integration is no longer given. Thus, this shortcoming with reload is venial.
Good luck!

**Reply by juerg-altwegg** [reply]
Found another integration with the exact same problem: https://github.com/thomasloven/hass-plejd/issues/53

**Reply by juerg-altwegg** [reply, marked as answer]
Finally, the solution is found :-)
As mentioned by SukramJ the state changes _from_ or _to_ "unknown". Thus, the YAML-Code for the triggered event can be supplemented with "not_from" and "not_to" as follows:

```
platform: state
entity_id:
  - event.tastenwippe_hm_pb_2_wm55_2_ch1
not_from:
  - unknown
  - unavailable
not_to:
  - unknown
  - unavailable
```

More details can be found here: https://community.home-assistant.io/t/help-with-triggering-an-automation-only-when-date-is-changed-in-the-state/630111

And it works: a reload doesn't trigger such configured actions/events.

**Comment by SukramJ** [comment]
Others will have the same problem, so i reopen this.
